File: Data Structures & Algorithms/generate-parentheses/submission-2.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:
        res=[]
        start="("
        def generateAll(st,openP,closeP,m):
            if len(st) > m or closeP>openP:
                return
            if openP+closeP==m and openP==closeP:
                res.append(st)
                return
            generateAll(st+")",openP,closeP+1,m)
            generateAll(st+"(",openP+1,closeP,m)
        generateAll(start,1,0,n*2);
        return res

File: Data Structures & Algorithms/longest-palindromic-substring/submission-2.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int res=0,size=s.length();
        int start=0;
        for (int i =0;i<size;i++){
            int l=i,r=i;
            while (l>=0 && r<size && s[l]==s[r]){
                 if (r-l+1>res){
                res=r-l+1;
                start=l;
            }
                l--;
                r++;
            }
           
            l=i,r=i+1;
            while (l>=0 && r<size && s[l]==s[r]){
                 if (r-l+1>res){
                res=r-l+1;
                start=l;
            }
                l--;
                r++;
            }
        
        }
        return s.substr(start,res);
    }
};
